src/uwtools/tests/drivers/test_filter_topo.py

```diff
@@ -0,0 +1,102 @@
+# pylint: disable=missing-function-docstring,protected-access,redefined-outer-name
+"""
+filter_topo driver tests.
+"""
+from pathlib import Path
+from unittest.mock import DEFAULT as D
+from unittest.mock import patch
+
+import f90nml  # type: ignore
+from iotaa import refs
+from pytest import fixture
+
+from uwtools.config.support import from_od
+from uwtools.drivers.driver import Driver
+from uwtools.drivers.filter_topo import FilterTopo
+
+# Fixtures
+
+
+@fixture
+def config(tmp_path):
+    input_grid_file = tmp_path / "C403_grid.tile7.halo4.nc"
+    input_grid_file.touch()
+    return {
+        "filter_topo": {
+            "config": {
+                "input_grid_file": str(input_grid_file),
+            },
+            "execution": {
+                "executable": "/path/to/orog_gsl",
+            },
+            "namelist": {
+                "update_values": {
+                    "filter_topo_nml": {
+                        "grid_file": "/path/to/grid/file",
+                        "grid_type": 0,
+                        "mask_field": "land_frac",
+                        "regional": True,
+                        "res": 403,
+                        "stretch_fac": 0.999,
+                        "topo_field": "orog_filt",
+                        "topo_file": "/path/to/topo/file",
+                        "zero_ocean": True,
+                    }
+                }
+            },
+            "run_dir": str(tmp_path / "run"),
+        }
+    }
+
+
+@fixture
+def driverobj(config):
+    return FilterTopo(config=config, batch=True)
+
+
+# Tests
+
+
+def test_FilterTopo():
+    for method in [
+        "_driver_config",
+        "_resources",
+        "_run_via_batch_submission",
+        "_run_via_local_execution",
+        "_runcmd",
+        "_runscript",
+        "_runscript_done_file",
+        "_runscript_path",
+        "_scheduler",
+        "_taskname",
+        "_validate",
+        "_write_runscript",
+        "run",
+        "runscript",
+    ]:
+        assert getattr(FilterTopo, method) is getattr(Driver, method)
+
+
+def test_FilterTopo_input_grid_file(driverobj):
+    path = Path(driverobj._driver_config["run_dir"]) / "C403_grid.tile7.halo4.nc"
+    assert not path.is_file()
+    driverobj.input_grid_file()
+    assert path.is_symlink()
```

Comment:
I must have skipped over it in the driver....this file gets linked? It does not end up getting overwritten by the program, right?

Reply:
The program modifies the topo (aka orography) file, not this one. From [the docs](https://noaa-emcufs-utils.readthedocs.io/en/latest/ufs_utils.html#id49):

> The filtered orography is written to the “orog_filt” record of the input orography file ...

If you look at the example run, the symlink `C403_grid.tile7.halo6.nc` in the run directory is symlinked to a file that you own, that I cannot write to, so I'm confident that it's not the one being written to.